src/compiler/parser.ts

```diff
@@ -3210,6 +3271,215 @@ module ts {
             node.name = parseRightSideOfDot(/*allowIdentifierNames*/ true);
             return finishNode(node);
         }
+        
+        // Return a string representation of an EntityName
+        function entityNameToString(entityName: EntityName): string {
+            if (entityName.kind === SyntaxKind.Identifier) {
+                return (<Identifier>entityName).text;
+            } else {
+                let qualifiedName = <QualifiedName>entityName;
+                return entityNameToString(qualifiedName.left) + '.' + entityNameToString(qualifiedName.right);
+            }
+            return "";
+        }
+
+        function hasClosingTagAfterPos(tagName: string, currentPos: number): boolean {
+            if (!closingTagsMap) {
+                closingTagsMap = retrieveClosingTagsMap(sourceText, sourceFile.languageVersion);

```

Comment:
Removed (this was part of the speculative parsing when we still allowed `<T>` type assertions)